hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiringCreations.java

```diff
@@ -24,20 +24,28 @@
 import com.hedera.services.ledger.HederaLedger;
 import com.hedera.services.records.RecordCache;
 import com.hedera.services.state.EntityCreator;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hedera.services.state.merkle.MerkleEntityId;
 import com.hedera.services.state.submerkle.ExpirableTxnRecord;
 import com.hederahashgraph.api.proto.java.AccountID;
 import com.hederahashgraph.api.proto.java.TransactionRecord;
+import com.swirlds.fcmap.FCMap;
+
+import java.util.function.Supplier;
 
 public class ExpiringCreations implements EntityCreator {
 	private RecordCache recordCache;
-	private HederaLedger ledger;
+
 	private final ExpiryManager expiries;
 	private final GlobalDynamicProperties dynamicProperties;
+	private final Supplier<FCMap<MerkleEntityId, MerkleAccount>> accounts;
 
 	public ExpiringCreations(
 			ExpiryManager expiries,
-			GlobalDynamicProperties dynamicProperties
+			GlobalDynamicProperties dynamicProperties,
```

Comment:
Do we also need a `Supplier` of `GlobalDynamicProperties`?
We should add an EET, if we didn't have already, to make sure a dynamic change of `dynamicProperties.cacheRecordsTtl()` does affect the expiry in the generated record.

Reply:
The types that need a `Supplier` are the children of `ServicesState` (since each time `copy()` is called the mutable instance changes). So e.g. `MerkleNetworkContext` **does** need a `Supplier` 😁 ...but `GlobalDynamicProperties` is always the same object. Only its fields change when file `0.0.121` is updated.

I added [`RecordCreationSuite.recordsTtlChangesAsExpected()` spec](https://github.com/hashgraph/hedera-services/blob/01422-M-SimplifyRecordExpiry/test-clients/src/main/java/com/hedera/services/bdd/suites/records/RecordCreationSuite.java#L209) to test dynamic changes to `cache.records.ttl`. But I left it commented out, to not run in CircleCI; reasons are in [this comment](https://github.com/hashgraph/hedera-services/blob/01422-M-SimplifyRecordExpiry/test-clients/src/main/java/com/hedera/services/bdd/suites/records/RecordCreationSuite.java#L84); specifically,
1. Because the expiration queue [requires its entries](https://github.com/hashgraph/hedera-services/blob/01422-M-SimplifyRecordExpiry/hedera-node/src/main/java/com/hedera/services/state/expiry/MonotonicFullQueueExpiries.java#L38) to have monotonic increasing expiration times, the EET spec has to begin by sleeping for 180s.
2. The records in state are used for duplicate classification, which protects against someone resubmitting a transaction from the record stream. This works because the [maximum valid lifetime](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/resources/bootstrap.properties#L55) of a transaction is also 180s. Because of how sensitive these values are, and how much they shape client expectations, it is **highly** unlikely we will ever make a dynamic change to the `cache.records.ttl`+`hedera.transaction.maxValidDuration` pair.

Reply:
Thanks for adding the spec. I agree with you that we don't need to run it in CI. I especially agree with you that there is a very small chance that we will dynamically update the `(cache.records.ttl, hedera.transaction.maxValidDuration)` pair. Just having the spec and running it locally/manually, to make sure a dynamic change of dynamicProperties.cacheRecordsTtl() does affect the expiry in the generated record, is enough.